apps/heft/src/plugins/CopyStaticAssetsPlugin.ts

```diff
@@ -13,7 +15,56 @@ import { CopyFilesPlugin } from './CopyFilesPlugin';
 
 const PLUGIN_NAME: string = 'CopyStaticAssetsPlugin';
 
+interface IPartialTsconfigCompilerOptions {
+  outDir?: string;
+}
+
+interface IPartialTsconfig {
+  compilerOptions?: IPartialTsconfigCompilerOptions;
+}
+
 export class CopyStaticAssetsPlugin extends CopyFilesPlugin {
+  private static __partialTsconfigFileLoader: ConfigurationFile<IPartialTsconfig> | undefined;
+
+  private static get _partialTsconfigFileLoader(): ConfigurationFile<IPartialTsconfig> {
+    if (!CopyStaticAssetsPlugin.__partialTsconfigFileLoader) {
+      const schemaPath: string = path.resolve(__dirname, '..', 'schemas', 'anything.schema.json');
+      CopyStaticAssetsPlugin.__partialTsconfigFileLoader = new ConfigurationFile<IPartialTsconfig>({
+        projectRelativeFilePath: 'tsconfig.json',
+        jsonSchemaPath: schemaPath,
+        propertyInheritance: {
+          compilerOptions: {
+            inheritanceType: InheritanceType.custom,
+            inheritanceFunction: (
+              currentObject: IPartialTsconfigCompilerOptions | undefined,
+              parentObject: IPartialTsconfigCompilerOptions | undefined
+            ) => {
+              if (currentObject && !parentObject) {
+                return currentObject;
+              } else if (!currentObject && parentObject) {
+                return parentObject;
+              } else if (parentObject && currentObject) {
+                return {
+                  ...parentObject,
+                  ...currentObject
+                };
+              } else {
+                return undefined;
+              }
+            }
+          }
+        },
+        jsonPathMetadata: {
+          '$.compilerOptions.outDir': {
+            pathResolutionMethod: PathResolutionMethod.resolvePathRelativeToConfigurationFile
```

Comment:
Okay this is pretty cool didn't know we had this functionality

Reply:
Yeah it's pretty useful